app/components/UserValidator/index.js

```diff
@@ -56,39 +59,64 @@ class Validator extends Component<Props, State> {
         }
       )
       .then(() => {
-        if (this.input) {
-          this.input.focus();
+        if (input) {
+          input.focus();
         }
       });
   };
 
-  render() {
-    return (
-      <div>
-        <div
-          className={cx(styles.overlay, {
-            [styles.shown]: this.state.showCompleted,
-          })}
-        >
-          <h3>
-            Tusen takk! Kos deg{' '}
-            <span role="img" aria-label="smile">
-              😀
-            </span>
-          </h3>
-          <i className="fa fa-check" />
-        </div>
-        <SearchPage
-          {...this.props}
-          placeholder="Skriv inn brukernavn eller navn"
-          handleSelect={this.handleSelect}
-          inputRef={(input) => {
-            this.input = input;
+  return (
+    <div>
+      <div
+        className={cx(styles.overlay, {
+          [styles.shown]: completed,
+        })}
+      >
+        <h3>
+          Tusen takk! Kos deg{' '}
+          <span role="img" aria-label="smile">
+            😀
+          </span>
+        </h3>
+        <i className="fa fa-check" />
+      </div>
+      <Button
+        className={styles.scannerButton}
+        onClick={() => setShowScanner(true)}
+      >
+        Vis Scanner
+      </Button>
+      <Modal show={showScanner} onHide={() => setShowScanner(false)}>
+        <QrReader
+          onResult={(res, error) => {
+            if (!processing && res) {
+              handleSelect({
+                username: res.getText(),
+                result: '',
+                color: '',
+                content: '',
+                icon: '',
+                label: '',
+                link: '',
+                path: '',
+                picture: '',
+                value: '',
+              });
+            }
           }}
```

Comment:
Do we need some sort of error handling here (or atleast a console log)? Could be an idea to just add a pop up or something, any kind of error feed back is nice.
```suggestion
          onResult={(res, error) => {
            if (!processing && res) {
              handleSelect({
                username: res.getText(),
                result: '',
                color: '',
                content: '',
                icon: '',
                label: '',
                link: '',
                path: '',
                picture: '',
                value: '',
              });
            }
            if (!!error) {
                console.info(error)
            }
          }}
```

Reply:
Sure, should not really matter tho, as when it throws an error, the scanner will prob not be displayed. I added it